File: src/java/eu/sig/training/ch11/DeadCode.java
package eu.sig.training.ch11;

public class DeadCode {
    public class Transaction {
        public Transaction(@SuppressWarnings("unused") long uid) {
        }
    }

    @SuppressWarnings("unused")
    // tag::getTransaction[]
    public Transaction getTransaction(long uid) {
        Transaction result = new Transaction(uid);
        if (result != null) {
            return result;
        } else {
            return lookupTransaction(uid); // <1>
        }
    }
    // end::getTransaction[]

    private Transaction lookupTransaction(
        @SuppressWarnings("unused") long uid) {
        return null;
    }
}

File: src/csharp/eu/sig/training/ch11/DeadCode.cs
namespace eu.sig.training.ch11
{

    public class DeadCode
    {
        public class Transaction
        {
            public Transaction(long uid)
            {
            }
        }

        // tag::getTransaction[]
        public Transaction GetTransaction(long uid)
        {
            Transaction result = new Transaction(uid);
            if (result != null)
            {
                return result;
            }
            else
            {
                return LookupTransaction(uid); // <1>
            }
        }
        // end::getTransaction[]

        private Transaction LookupTransaction(long uid)
        {
            return null;
        }
    }
}
